### PR TITLE
haproxyloadbalancer: Ensure we don't have transactions open

### LINKED ISSUE #891: HAProxy LoadBalancer transactions are not closed
/kind bug

**What steps did you take and what happened:**
Created a CAPI cluster with HAProxy loadbalancer.

Looked at the HAProxy transactions and noticed a lot of transactions that are not closed,  resulting in stale transactions that are in progress

**What did you expect to happen:**
Transactions should be closed

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

`curl -s -k -X GET -u "<username>:<password>" --cacert ca.crt --cert client.crt --key client.key https://<haproxy_ip>:<port>/v1/services/haproxy/transactions  | jq '[ .[] | select(.status | contains("in_progress")) ] | length'
===> 439`

**Environment:**

- Cluster-api-provider-vsphere version: v0.6.3
- Kubernetes version: (use `kubectl version`): 1.17.3
- OS (e.g. from `/etc/os-release`): 
NAME="VMware Photon OS"
VERSION="3.0"
ID=photon
VERSION_ID=3.0
PRETTY_NAME="VMware Photon OS/Linux"
ANSI_COLOR="1;34"
HOME_URL="https://vmware.github.io/photon/"
BUG_REPORT_URL="https://github.com/vmware/photon/issues"

### PR DESCRIPTION
**Pull request by randomvariable**
Signed-off-by: Naadir Jeewa <jeewan@vmware.com>

**What this PR does / why we need it**:
Ensure we commit transactions, and additionally delete any open transactions on re-entrancy.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #891 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
* Ensure HAProxy transactions are always closed.
```

### CODE REVIEW

File: controllers/haproxyloadbalancer_controller.go [hunk 1]
```diff
@@ -414,12 +414,25 @@ func (r haproxylbReconciler) reconcileLoadBalancerConfiguration(ctx *context.HAP
 		return errors.Wrap(err, "Failed to get HAProxy dataplane global config")
 	}
 
+	// Attempt to close existing transactions in case controller had previously crashed
+	if err := r.closeAllTransactions(ctx, client); err != nil {
```
Comment (yastij) [inline controllers/haproxyloadbalancer_controller.go:418]: doing this here might also close on-going transactions of other `HAProxyLoadbalancers` nope ?

Comment (randomvariable) [reply controllers/haproxyloadbalancer_controller.go:418]: Only on that same VM instance. Given that we're currently string templating everything anyway, other controllers are going to have problems reconfiguring the LB anyway. 

We could revisit this when we go to Dataplane API v2 and persist a transaction ID back to the resource to deal with re-entrancy.

Comment (yastij) [reply controllers/haproxyloadbalancer_controller.go:418]: @randomvariable - can you add a `TODO` here ?

Comment (randomvariable) [reply controllers/haproxyloadbalancer_controller.go:418]: Will do

Comment (randomvariable) [reply controllers/haproxyloadbalancer_controller.go:421]: Sorry for the lateness. Updated, and then will follow up with a backport PR to release-0.6

**Pull request merged**